plugins/Live/VisitorDetails.php

```diff
@@ -309,4 +308,24 @@ public static function getReferrerSummaryForVisit($visit)
 
         return $visit->getColumn('referrerName');
     }
+
+    /**
+     * @return array
+    */
+    private function getAdditionalUrlsForSite(): array
+    {
+        $cache = Cache::getTransientCache();
+        $cacheKey = 'Live.additionalSiteUrls.' . $this->getIdSite();
+
+        if (!$cache->contains($cacheKey)) {
+            $sitesModel = new \Piwik\Plugins\SitesManager\Model();
+
+            $cache->save(
+                $cacheKey,
+                $sitesModel->getAliasSiteUrlsFromId($this->getIdSite())
+            );
```

Comment:
Saves the extra fetch when not needed.

```suggestion
            $sitesModel = new \Piwik\Plugins\SitesManager\Model();
            $aliasSiteUrls = $sitesModel->getAliasSiteUrlsFromId($this->getIdSite());

            $cache->save(
                $cacheKey,
                $aliasSiteUrls
            );
            
            return $aliasSiteUrls;
```

Reply:
The block was actually designed to look like most of `if (!$cache->contains())` constructs I found, with an explicit `$cache->fetch()` after it.

The other choice I found was `if ($cache->contains())` with an early return at the beginning, and having the `$cache->save(); return` without nesting.

Or a simple `if {} else {}; return $var`.

Happy with every option here.

Reply:
Right. I always forget there could pre-existing patterns we duplicate/tweak, and then I'm on the fence with whether we should keep doing that or do it better/differently in the areas we're touching.

Perhaps if there's a pattern with cache contains and an early return,  we can do that. I'm not sure the explicit fetch has any meaning here apart from possibly having some logic outside in the cache layer that would, I don't know, amend the value somehow 🤷.

Plus, if the write fails, it would then fetch nothing, so returning the currently processed value anyway makes it a bit more robust?

Reply:
Doing it differently could be a nice `$cache->get($key, function () { /* generate value */ });` :sweat_smile: 

And here I'll opt for the early return as a "cache short-circuits execution" pattern.